Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/DialogFragmentTemplate.kt

```diff
@@ -0,0 +1,41 @@
+package de.rki.coronawarnapp.ui.dialog
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.ContextExtensions.getColorCompat
+
+class DialogFragmentTemplate(
+    private val cancelable: Boolean = true,
+    private val isDeleteDialog: Boolean = false,
+    private val dismissAction: () -> Unit = { },
+    private val materialDialog: MaterialAlertDialogBuilder? = null,
+    private val config: MaterialAlertDialogBuilder.() -> Unit
+) : DialogFragment() {
```

Comment:
I'm not sure about passing arguments to fragments from the constructor, I think the best way is to use a Bundle 
https://developer.android.com/reference/android/app/DialogFragment

Reply:
Ok,  I've removed the constructor params and created a parcelized data class to pass them as a bundle like we do with the `RecycleBinDialogFragment`